bump some utility dependencies;  disable renovate major bumps for packages with newer ESM-only versions

legobeat opened the pull request:
Bumps the following dependencies:

#### `lavamoat-viz`
- Bump `open` from `7.4.2` to `8.4.2`
  - Holding back due to v9+ being ESM-only
  - Replaces #936 (rf for references)
- Bump `pify` from `4.0.1` to `5.0.0`
  - Holding back due to v6+ being ESM-only
  - Replaces #938 (rf for references)

#### `lavamoat-browserify`
- Bump `pify` from `4.0.1` to `5.0.0`
  - Holding back due to v6+ being ESM-only
  - Replaces #938 (rf for references)

#### `survey`
- Bump `p-limit` from `2.3.0` to `3.1.0`
  - Holding back due to v4+ being ESM-only
  - Replaces #937 (rf for references)

#### Tooling
- renovate: Ignore new major versions for packages with new ESM-only versions

legobeat committed:
fix(deps/viz): open@7.4.2->8.4.2

legobeat committed:
fix(deps/survey): p-limit@2.3.0->3.1.0

legobeat committed:
fix(deps/browserify,viz): pify@4.0.1->5.0.0

legobeat committed:
chore(browserify): commit changes from test:prep

legobeat committed:
chore(deps): npm dedupe; npm audit fix; npm run test:prep

boneskull commented:
@legobeat Going to push an update to the renovate config here, as it will attempt to upgrade these modules further if we don't disallow them.

boneskull committed:
chore(renovate): no major bumps of ESM-only pkgs

This creates a list in the Renovate config which exempts ESM-only packages from major upgrades.

Such packages will _mostly_ fail as long as we are _mostly_ a CJS codebase; we can upgrade majors manually otherwise.

boneskull commented on .github/renovate.json5:
```diff
@@ -9,5 +9,43 @@
     ':semanticCommits', // use semantic commits
     'group:linters', // group lint-related packages together
   ],
+  packageRules: [
```
I need to reformat this...

boneskull reviewed (approved):
lgtm; see my change to renovate

boneskull commented:
Ref: #962
